file_handler.cpp

```diff
@@ -70,25 +91,27 @@ void file_handler::handle_request(const request &req, reply& rep){
 	std::string extension;
  	if (last_dot_pos != std::string::npos && last_dot_pos > last_slash_pos)
 	{
-	extension = request_path.substr(last_dot_pos + 1);
+		extension = request_path.substr(last_dot_pos + 1);
 	}
 
 	std::string full_path = docroot + request_path;
 	std::ifstream is(full_path.c_str(), std::ios::in | std::ios::binary);
 	if (!is)
 	{
-	rep = reply::stock_reply(reply::not_found);
-	return;
+		*response = Response::stock_response(Response::not_found);
+		return RequestHandler::not_found;
 	}
 
-	rep.status = reply::ok;
+	response->SetStatus(Response::ok);
+
+	std::string body;
 	char buf[512];
 	while (is.read(buf, sizeof(buf)).gcount() > 0)
-rep.content.append(buf, is.gcount());
-	rep.headers.resize(2);
-	rep.headers[0].name = "Content-Length";
-	rep.headers[0].value = std::to_string(rep.content.size());
-	rep.headers[1].name = "Content-Type";
+		body.append(buf, is.gcount());
+
```

Comment:
umm idk i only refactored that part but unlike the echo_handler version this works at least :)